experimental/FTheoryTools/TateModels/constructors.jl

```diff
@@ -0,0 +1,201 @@
+
+julia> a4 = sum([rand(Int) * b for b in basis_of_global_sections(anticanonical_bundle(base)^4)]);
+
+julia> a6 = sum([rand(Int) * b for b in basis_of_global_sections(anticanonical_bundle(base)^6)]);
+
+julia> t = global_tate_model([a1, a2, a3, a4, a6], base)
+Global Tate model over a concrete base
+
+julia> is_smooth(toric_ambient_space(t))
+false
+```
+"""
+function global_tate_model(ais::Vector{T}, base::AbstractNormalToricVariety) where {T<:MPolyElem{fmpq}}
+    if length(ais) != 5
+        throw(ArgumentError("We require exactly 5 Tate sections"))
+    end
+    if any(k -> parent(k) != cox_ring(base), ais)
+        throw(ArgumentError("All Tate sections must reside in the Cox ring of the base toric variety"))
+    end
+    toric_ambient_space = _ambient_space_from_base(base)
+    pt = _tate_polynomial(ais, cox_ring(toric_ambient_space))
+    Y4 = closed_subvariety_of_toric_variety(toric_ambient_space, [pt])
+    model = GlobalTateModel(ais[1], ais[2], ais[3], ais[4], ais[5], pt, base, toric_ambient_space, Y4)
+    set_attribute!(model, :base_fully_specified, true)
+    return model
+end
+
+
+################################################
+# 3: Constructors over not fully specified bases
+################################################
+
+@doc Markdown.doc"""
+    global_tate_model(ais::Vector{T}, auxiliary_base_ring::MPolyRing, d::Int) where {T<:MPolyElem{fmpq}}
+
+This method constructs a global Tate model over a base space that is not
+fully specified. The following example exemplifies this approach.
+
+# Examples
+```jldoctest
+julia> auxiliary_base_ring, (a10, a21, a32, a43, a65, w) = QQ["a10", "a21", "a32", "a43", "a65", "w"];
+
+julia> a1 = a10;
+
+julia> a2 = a21 * w;
+
+julia> a3 = a32 * w^2;
+
+julia> a4 = a43 * w^3;
+
+julia> a6 = a65 * w^5;
+
+julia> ais = [a1, a2, a3, a4, a6];
+
+julia> t = global_tate_model(ais, auxiliary_base_ring, 3)
+Global Tate model over a not fully specified base
+
+julia> tate_polynomial(t)
+-a10*x*y*z + a21*w*x^2*z^2 - a32*w^2*y*z^3 + a43*w^3*x*z^4 + a65*w^5*z^6 + x^3 - y^2
+
+julia> toric_base_space(t)
+[ Info: Base space was not fully specified. Returning AUXILIARY base space.
+Normal toric variety
+
+julia> toric_ambient_space(t)
+[ Info: Base space was not fully specified. Returning AUXILIARY ambient space.
+Normal, simplicial toric variety
+
+julia> dim(toric_ambient_space(t))
+5
+```
+"""
+function global_tate_model(ais::Vector{T}, auxiliary_base_ring::MPolyRing, d::Int) where {T<:MPolyElem{fmpq}}
+    if length(ais) != 5
+        throw(ArgumentError("We expect exactly 5 Tate sections"))
+    end
+    if any(k -> parent(k) != auxiliary_base_ring, ais)
+        throw(ArgumentError("All Tate sections must reside in the provided auxiliary base ring"))
+    end
+    if d <= 0
+        throw(ArgumentError("The dimension of the base space must be positive"))
+    end
+    if length([string(k) for k in gens(auxiliary_base_ring)]) < d
```

Comment:
Again, isn't this just
```suggestion
    if ngens(auxiliary_base_ring) < d
```